### PR TITLE
UNT-T12274 Updated base class code and javadoc

### PR DESCRIPTION
**Pull request by vraj-s-shah**
Updated the javadoc for all the functions and optimised code in base classes wherever required. Updated few functions to make sure they follow SRP.

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt [hunk 1]
```diff
@@ -14,280 +14,287 @@ import {{ cookiecutter.package_name }}.BR
...
-        val binding: ViewDataBinding = if (viewType == itemTypeNormal) {
-            DataBindingUtil.inflate(LayoutInflater.from(parent.context), getLayoutIdForType(viewType), parent, false)
-        } else {
-            DataBindingUtil.inflate(LayoutInflater.from(parent.context), getLayoutIdForLoading(viewType), parent, false)
+    /**
+     * Inner class to set recycler view holder
+     *
+     * @param viewDataBinding   The [ViewDataBinding] instance
+     */
+    inner class RecyclerHolder(
```
Comment (prem-p-simform) [inline {{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt:30]: Put these inner classes at the very beginning of the `BaseRecyclerAdapter` class to group properly

**Review by prem-p-simform — CHANGES_REQUESTED** [review]

File: {{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt (hunk 1 above)
Comment (vraj-s-shah) [reply {{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt:30]: @prem-p-simform They are already at the beginning in my opinion, as there are only variables above them. Do you want to move these inner classes above these variables?

Comment (prem-p-simform) [reply {{ cookiecutter.repo_name }}/app/src/main/java/{{ cookiecutter.package_dir }}/ui/base/BaseRecyclerAdapter.kt:30]: @vraj-s-shah Yes please move above variables

**Comment by nikunj-b-simform** [comment]
@vraj-s-shah Make it a practise to always include the **description** when you increase the PR.

**Review by prem-p-simform — APPROVED** [review]

**Pull request merged**